torch/_inductor/decomposition.py

```diff
@@ -261,13 +265,13 @@ def bmm(
     batch2: torch.Tensor,
 ) -> torch.Tensor:
     if config.coordinate_descent_tuning and self.device.type != "cpu":
-        if guard_size_oblivious(self.shape[1] == 1) or guard_size_oblivious(
+        if guard_else_false(self.shape[1] == 1) or guard_else_false(
             batch2.shape[2] == 1
         ):
             out = (self.unsqueeze(-1) * batch2.unsqueeze(1)).sum(dim=2)
             return out
     if self.device.type == "cpu":
-        if guard_size_oblivious(self.size(1) == 1) and guard_size_oblivious(
+        if guard_else_false(self.size(1) == 1) and guard_else_false(
             batch2.size(-1) == 1
         ):
```

Comment:
Since these are all optimizations, we should just be doing statically_known_true. or potentially an api that is only willing to add guards on non-export use cases.

Reply:
mhmm this mostly keep the semantics of guard_size_oblivious, 
moving to statically_known_true will change the behaviour. 
i will start failing with unbacked.

this API does what you explained, just for unbacked vs backed instead of export vs non export.

Reply:
@laithsakka I think what @eellison is saying is that in these cases we *should* change the behavior and move to `statically_known_true` even for the case of unbacked. Because these are optimizations, we don't want to guard, even if we can.

Reply:
i see make sense we can address that in different PR maybe? or I can put PR for that and rebase this on top of it .